### PR TITLE
[NOTEPAD] Improve printing (header/footer/tabs/performance)

### PR DESCRIPTION
**Pull request by katahiromz**
## Purpose
Improve the printing feature.
JIRA issue: [CORE-18837](https://jira.reactos.org/browse/CORE-18837)

## Proposed changes

- Fix `GetPrintingRect` function.
- Add `DrawHeaderOrFooter`, `DoPrintBody`, `DoPrintPage`, `DoCreatePrintFonts`, and `DoPrintDocument` helper functions.
- Implement page header/footer and tabs printing in `DIALOG_FilePrint` function.
- Improve printing performance.
- Modify the header/footer resource strings.
- Delete the needless resource strings.

## TODO

- [x] Do tests.

### CODE REVIEW

File: base/applications/notepad/dialog.c [hunk 1]
```diff
@@ -283,24 +283,20 @@ int GetSelectionText(HWND hWnd, LPTSTR lpString, int nMaxCount)
...
-    iLogPixelsX = GetDeviceCaps(hdc, LOGPIXELSX);
-    iLogPixelsY = GetDeviceCaps(hdc, LOGPIXELSY);
-    iHorzRes = GetDeviceCaps(hdc, HORZRES);
-    iVertRes = GetDeviceCaps(hdc, VERTRES);
-
-    rcPrintRect.left = (margins.left * iLogPixelsX / 2540) - iPhysPageX;
-    rcPrintRect.top = (margins.top * iLogPixelsY / 2540) - iPhysPageY;
-    rcPrintRect.right = iHorzRes - (((margins.left * iLogPixelsX / 2540) - iPhysPageX) + ((margins.right * iLogPixelsX / 2540) - (iPhysPageW - iPhysPageX - iHorzRes)));
-    rcPrintRect.bottom = iVertRes - (((margins.top * iLogPixelsY / 2540) - iPhysPageY) + ((margins.bottom * iLogPixelsY / 2540) - (iPhysPageH - iPhysPageY - iVertRes)));
+#define CONVERT_X(x) MulDiv((x), iLogPixelsX, 2540) /* 100th millimeters to pixels */
```
Comment (learn-more) [inline base/applications/notepad/dialog.c:292]: Any reason why you left out `PHYSICALOFFSETX` ?

Comment (katahiromz) [reply base/applications/notepad/dialog.c:292]: I thought these are needless in actual printing.
But the physical margins should seem be checked before printing. The logical margins of the current settings should be larger than the physical margins.

Comment (katahiromz) [reply base/applications/notepad/dialog.c:294]: Adjusted.

**Comment by HBelusca** [comment]
Has this PR been reviewed and approved before being merged?

**Comment by katahiromz** [comment]
Many times ChatGPT reviewed this PR.

**Pull request merged**